transformer_engine/pytorch/csrc/ts_fp8_op.cpp

```diff
@@ -0,0 +1,179 @@
+/*************************************************************************
+ * Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include <torch/script.h>
+#include "extensions.h"
+
+transformer_engine::DType reverse_map_dtype(int64_t dtype) {
```

Comment:
Hmm, can we make it just a static_cast + boundary condition ensuring that it is valid? While I don't expect adding new types very often, this switch is a potential source of problems if we do add a new type.

Reply:
Also, could you put this function in an unnamed namespace maybe?

Reply:
I have taken care of the remaining comments locally and will push the changes shortly. 

> can we make it just a static_cast + boundary condition ensuring that it is valid

Quick question on this comment - are you suggesting that the switch statement to be removed? Can you elaborate on how we can achieve the int to dtype mapping in that case?

Reply:
Yeah, I thought about something like
```
if (dtype >= 0 && dtype < largest_value) {
  return static_cast<transformer_engine::DType>(dtype);
} else {
  ERROR
}
```
We would probably need to add then some `DType::Num_Types` type to the enum though.

Reply:
Ah I see what you mean - will make this change. I do see a `kNumTypes` in the enum already. Let me test it and push the change.

Reply:
Addressed both comments in the latest commits